### PR TITLE
[AlertDialog] Explicit modality

### PR DESCRIPTION
**Pull request by andy-hook**
Force `Modal` mode on `AlertDialog` and prevent dismiss on click outside.

### CODE REVIEW

File: packages/react/alert-dialog/src/AlertDialog.tsx [hunk 1]
```diff
@@ -14,8 +14,10 @@ import type * as Polymorphic from '@radix-ui/react-polymorphic';
 
 const ROOT_NAME = 'AlertDialog';
 
-const AlertDialog: React.FC<React.ComponentProps<typeof DialogPrimitive.Root>> = (props) => (
-  <DialogPrimitive.Root {...props} />
+type AlertDialogProps = Omit<React.ComponentProps<typeof DialogPrimitive.Root>, 'modal'>;
```
Comment (andy-hook) [inline packages/react/alert-dialog/src/AlertDialog.tsx:17]: Do I have this right? we want to enforce this opinion on consumers?

Comment (benoitgrelard) [reply packages/react/alert-dialog/src/AlertDialog.tsx:17]: Yeah I think so.

File: packages/react/alert-dialog/src/AlertDialog.tsx [hunk 2]
```diff
@@ -62,6 +64,9 @@ const AlertDialogContent = React.forwardRef((props, forwardedRef) => {
             event.preventDefault();
             cancelRef.current?.focus({ preventScroll: true });
           })}
+          onPointerDownOutside={composeEventHandlers(contentProps.onPointerDownOutside, (event) =>
```
Comment (benoitgrelard) [inline packages/react/alert-dialog/src/AlertDialog.tsx:67]: I think we also need to prevent the escape keydown one.

Comment (benoitgrelard) [reply packages/react/alert-dialog/src/AlertDialog.tsx:67]: May be worth rejigging what's exposed from `Dialog` to facilitate the changes here also.

Comment (benoitgrelard) [reply packages/react/alert-dialog/src/AlertDialog.tsx:67]: Ha cool, good to know. I suppose that's still a meaningful user action, as opposed to clicking outside by mistake potentially.

Comment (benoitgrelard) [reply packages/react/alert-dialog/src/AlertDialog.tsx:67]: We probably should just hardcode it then and remove it from the types.

Comment (andy-hook) [reply packages/react/alert-dialog/src/AlertDialog.tsx:67]: > We probably should just hardcode it then and remove it from the types.

That's fine with me, are we sure consumers won't need access to the handler for a different purpose?

**Pull request merged**